ddtrace/appsec/_processor.py

```diff
@@ -413,3 +415,20 @@ def on_span_finish(self, span: Span) -> None:
         finally:
             # release asm context if it was created by the span
             _asm_request_context.unregister(span)
+
+            if span.span_type != SpanTypes.WEB:
+                return
+
+            to_delete = []
+            for iterspan, ctx in self._span_to_waf_ctx.items():
+                # delete all the ddwaf ctxs associated with this span or finished or deleted ones
+                if iterspan == span or iterspan.finished:
+                    # so we don't change the dictionary size on iteration
+                    to_delete.append(iterspan)
+
+            for s in to_delete:
+                try:
+                    del self._span_to_waf_ctx[s]
+                    del s
```

Comment:
isn't this deletig the span?

Reply:
Just the weak ref, but it was an experiment, removed.